signup.js

```diff
@@ -0,0 +1,82 @@
+const signupBtn = document.getElementById('signup-button');
+const form = document.getElementById('Form'); 
+const emailField = document.querySelector(".email-field");
+const emailInput = document.querySelector(".email");
+const passInput = document.getElementById('password');
+const eyeIcons = document.querySelectorAll(".visibility_off"); 
+const passField = document.querySelector(".create-password");
+const cPassInput = document.getElementById('password-repeat');
+const confirmError = document.querySelector(".confirm-error");
+
+eyeIcons.forEach(eyeIcon => {
+  eyeIcon.addEventListener("click", () => {
+    const passwordInput = eyeIcon.parentElement.querySelector("input");
+
+    if (passwordInput.type === "password") {
+      passwordInput.type = "text"; 
+      eyeIcon.src = "image/visibility_on_btn.png";
+    } else {
+      passwordInput.type = "password"; 
+      eyeIcon.src = "image/visibility_off_btn.png"; 
+    }
+  });
+});
+
+function checkEmail() {
+  const emailPattern = /^[^ ]+@[^ ]+\.[a-z]{2,3}$/;
+
+  if (!emailInput.value.match(emailPattern)) {
+    emailField.classList.add("invalid"); 
+    emailField.querySelector(".email-error").style.display = "block";
+  } else {
+    emailField.classList.remove("invalid"); 
+    emailField.querySelector(".email-error").style.display = "none";
+  }
+}
+
+function createPass() {
+  const passPattern = /^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)(?=.*[@$!%*?&])[A-Za-z\d@$!%*?&]{8,}$/;
+
+  if (!passInput.value.match(passPattern) || passInput.value.length < 8) {
+    passInput.classList.add("invalid");
+    passField.querySelector(".password-error").style.display = "block";
+  } else {
+    passInput.classList.remove("invalid");
+    passField.querySelector(".password-error").style.display = "none";
+  }
+}
+
+function checkPasswordMatch() {
+  const passValue = passInput.value;
+
+  if (passValue !== cPassInput.value) {
+    cPassInput.classList.add("invalid");
+    confirmError.style.display = "block";
+  } else {
+    cPassInput.classList.remove("invalid");
+    confirmError.style.display = "none";
+  }
+}
+
+form.addEventListener("submit", function(e) {
+  e.preventDefault();
+  checkEmail(); 
+  createPass();
+  checkPasswordMatch();
+});
+
+function movePage() {
+  checkEmail();
+  createPass();
+
+  if (emailField.classList.contains('invalid') || passInput.classList.contains('invalid')) {
+    return;
+  }
+
+  location.replace("login.html");
+}
```

Comment:
이 함수역시, movePage지만 movePage하지 않는 경우도 있다는 점, 이 자체의 내용상 "이메일을 check하고, Pass를 create하는 데, invalid클래스가 있으면 그냥 함수를 끝낸 다"라고 놓고 보면, 어떤 일을 하는 함수인지, 어떤 의도로 만든 함수인지 단번에 알기어려우니, 조금 더 의도를 드러낼 수 있겠다 는 생각이 듭니다.

```suggestion
function handleClickSignup() {
  if(!validateEmail()) { // 검증로직에 관한 함수 
    showEmailErrorMessage() // ui 변화에 관한 함수
    return; // 검증에서 실패하면 함수 끝내기
  }
  // password check 생략

  location.replace("login.html");
}
```